src/statsmodels/contrasts.jl

```diff
@@ -117,6 +117,11 @@ function ContrastsMatrix{C <: AbstractContrasts}(contrasts::C, levels::Vector)
 end
 
 ContrastsMatrix(C::AbstractContrasts, v::PooledDataArray) = ContrastsMatrix(C, levels(v))
+ContrastsMatrix{C <: AbstractContrasts}(c::Type{C}, col::PooledDataArray) = ContrastsMatrix(c(), col)
+ContrastsMatrix(c::ContrastsMatrix, col::PooledDataArray) =
+    isempty(symdiff(c.levels, levels(col))) ?
```

Comment:
The ternary operator isn't the best solution when you need to add line breaks. You can write this as:
```julia
isempty(symdiff(c.levels, levels(col))) || error(...)
return c
```

Also, in other places you explicitly list the mismatched contrasts.